main.cpp

```diff
@@ -31,6 +35,12 @@ TCB threads[1000];
  */
 int num_threads = 0;
 
+/**
+ * Ready List, waiting, suspended list
+ */
+std::vector<int> ready_list;
+std::vector<int> waiting_list;
+std::vector<int> suspended_list;
```

Comment:
The list of suspended threads could probably be in the same list as the waiting threads, but since waiting threads and suspended threads are technically not the same thing, I put them in different lists.